### DISCUSSION TITLE
How do I do conditional YAML frontmatter, e.g. based on profile?

### DISCUSSION THREAD

**Discussion by pchr8** [quarto-dev/quarto-cli#10064] [Q&A] [answered]
### Description

Use-case: better multilingual support is my current problem, but I think would be useful in many cases.

Currently, one can do per-language alternates, as shown in [the documentation](https://quarto.org/docs/authoring/language.html#per-language-alternates):

```qmd
---
title: "My Document"
author: "Norah Jones"
date: 5/22/2022
lang: fr
language:
  en:
    title-block-published: "Updated"
  fr:
    title-block-published: "Mis à jour"
---
```

Then one can set the language in `_quarto-en.yml` profile files, and it magically works. (I'm doing an approach similar to https://marioangst.com/de/blog/posts/multi-language-quarto/)

It would be really nice if one could also
```qmd
---
title: "My Document"
author: "Norah Jones"
date: 5/22/2022
lang: it
language:
  en:
    title: "My post number one"
  it:
    title: "Il mio post numero uno"
---
```

Or are there other ways to achive something similar?  I found no way to use variables inside shortcodes, basically something like `{{< meta titles[{{< env QUARTO_PROFILE >}}] >}}` won't work (and setting a separate meta variable for the language, and then in another front-matter variable referring to that value won't work as well).

I'm thinking about just having the three titles in the front-matter, and using a lua filter to pick one and set the "main" title based on it, but before I start reading into this — are there any easy solutions I'm missing?

**Comment by mcanouil** [comment]
If you are using profile, then you can set the title in it.
`language` is for variables used for translations, so there is not reason for `title` to be under `language`.

This being said, this has been extensively discussed already:
- current limitations on profile: https://github.com/quarto-dev/quarto-cli/issues/4516
- conditional content on format and meta: https://quarto.org/docs/authoring/conditional.html#matching-against-metadata
- multi lingual support: https://github.com/quarto-dev/quarto-cli/issues/275
- https://github.com/quarto-dev/quarto-cli/discussions/5262
- a duplicate or really close to your discussion: https://github.com/quarto-dev/quarto-cli/discussions/4455

I believe all these issues/discussions summarised various proof of concept and what is currently possible.

**Comment by pchr8** [comment, marked as answer]
Sorry, I spent hours researching this but somehow missed your first and last link which do answer my question, thank you!

I'll solve this with a Lua filter, checked it out and it's easier than it sounded initially. 

Thank you!

**Comment by pchr8** [comment]
This does what I want, basically, I'll swap prints for real logging etc. but otherwise I have a solution to the problem

```qmd
---
title: "Publications and Awards"
titles:
  en: "Publications and Awards"
  de: "Veröffentlichungen"
filters: [./filters/conditional-metadata.lua]

```

```lua
function Meta(m)
  local profiles = quarto.project.profile
  if not profiles then
    -- TODO: missing YAML key? Empty YAML key?..
    -- TODO even more later: filter multiple profiles to use the language one
    return m
  end

  local profile = profiles[1]
  -- If we have a named profile, save it, otherwise return
  if profile then
    print("Profile: " .. profile)
    m.active_profile = profile
  else
    return m
  end

  if m.titles then
    local titles = m.titles
    if titles[profile] then
      newtitle = pandoc.utils.stringify(titles[profile])
      oldtitle = pandoc.utils.stringify(m.title)
      -- log both if they differ
      if newtitle ~= oldtitle then
        m.title = newtitle
        -- print("Old title:" .. oldtitle)
        -- print("New title:" .. newtitle)
        print(oldtitle .. " => " .. newtitle)
      end
    else
      print("Title for profile " .. profile .. " not found among ")
      for lang, title in pairs(titles) do -- Table iteration.
        print("    " .. lang .. ": " .. pandoc.utils.stringify(title))
      end
    end
  end

  return m
end

```

**Reply by mcanouil** [reply]
Thanks for sharing what works for you!
